cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionTwillRunnerService.java

```diff
@@ -649,6 +653,7 @@ public TwillController create(@Nullable Callable<Void> startupTask, long timeout
               }
               programStateWriter.error(programRunId, throwable);
             }
+            restoreContext.cancel();
```

Comment:
Please use try..finally to ensure context is always properly cancelled, even if exception is thrown out. 

BTW: It's a perfect case for try with resources. We just need to either make Cancellable extend AutoClosable or at least setLoggingContext return an AutoClosable result

Reply:
Refactored the code to use try-with-resources. PTAL.